pygsti/tools/errgenproptools.py

```diff
@@ -459,6 +459,258 @@ def bch_approximation(errgen_layer_1, errgen_layer_2, bch_order=1, truncation_th
 
     return new_errorgen_layer_dict
 
+def magnus_expansion(errorgen_layers, magnus_order=1, truncation_threshold=1e-14):
+
+    """
+    Function for computing the nth-order magnus expansion for a set of error generator layers.
```

Comment:
Please add a reference for the math associated with this function (e.g., arXiv paper).

Please add type annotations, particularly for ``magnus_order: Literal[1,2,3]``.